index.js

```diff
@@ -17,100 +17,58 @@ let maxPage = 1;
 let page = 1;
 let searchQuery = "";
 
-async function fetchCharacters(searchQueryObject) {
-  const searchQueryString = searchQueryObject?.query;
-
+async function fetchCharacters() {
   try {
-    let url = "https://rickandmortyapi.com/api/character";
-
-    if (searchQueryString) {
-      cardContainer.innerHTML = "";
-      url += `?name=${encodeURIComponent(searchQueryString)}`;
-    }
-
-    if (!searchQueryString && page > 1) {
-      url += `?page=${page}`;
-    } else if (searchQueryString && page > 1) {
-      url += `&page=${page}`;
-    }
+    // We can always send the page parameter even if it is page 1 and also according to the instructions
+    // we can always send the searchQuery as well because the API will ignore if it is empty
+    let url = `https://rickandmortyapi.com/api/character?page=${page}&name=${encodeURIComponent(
+      searchQuery
+    )}`;
 
     const response = await fetch(url);
 
     if (response.ok) {
       const data = await response.json();
       const results = data.results;
-      let maxPage = data.info.pages; // Get the maxPage value from the response
+
+      maxPage = data.info.pages; // We can update the maxPage status directly
+      pagination.textContent = `${page} / ${maxPage}`; // Then we can update the pagination here
 
       cardContainer.innerHTML = "";
 
       results.forEach((result) => {
-        const options = {
-          image: result.image,
-          name: result.name,
-          status: result.status,
-          type: result.type,
-          occurrences: result.episode.length,
-        };
-
-        cardContainer.innerHTML += createCharacterCard(options);
+        cardContainer.innerHTML += createCharacterCard(result);
       });
-
-      // Update maxPage if a search query is provided
-      if (searchQueryString) {
```

Comment:
I change it to update the maxPage always, then it will always be the value from the api.